### PR TITLE
Feature/sc 28986/primehub feature support data collection

### PR DESCRIPTION
**Pull request by huaichehuang**
Signed-off-by: Dennis Huang dennishuang@infuseai.io

PrimeHub feature: support data collection script

PR checklist

- [X] Ensure you have added or ran the appropriate tests for your PR.
- [X] DCO signed
What type of PR is this?
New feature for PrimeHub troubleshooting

What this PR does / why we need it:
Provide a fast and convenient way to collect env info for troubleshooting PrimeHub.

Which issue(s) this PR fixes:
Collect all info including k8s, ceph and primehub env config and data.

Special notes for your reviewer:

Does this PR introduce a user-facing change?:
NONE

### CODE REVIEW

File: install/primehub-support [hunk 1]
```diff
@@ -0,0 +1,407 @@
...
+		echo "System memory usage is now logged at $LOG_DIR/ansible_mem_$TIMESTAMP.log"
+		echo "System disk usage is now logged at $LOG_DIR/ansible_disk_$TIMESTAMP.log"
+		echo "System partition table is now logged at $LOG_DIR/ansible_part_$TIMESTAMP.log"
+	fi
+else
+	echo "Ansible command not found, skipping....."
+	echo ""
+	SYS_CHECK
+fi
+echo ""
```
Comment (LiuYuWei) [inline install/primehub-support:243]: 這邊的程式碼和上面的程式碼是有重複的，不太確定為何是要這樣撰寫？

File: install/primehub-support [hunk 2]
```diff
@@ -0,0 +1,407 @@
...
+echo "Done."
+echo "-------------------------------------------------------------------------"
+echo ""
+
+echo "-------------------------------------------------------------------------"
+echo "Logs folder: $LOG_DIR"
+echo "Log archive: $(realpath $HOME/$FILE_NAME.tgz)"
+echo "Please send the log archive to support engineer for further analysis."
+echo "-------------------------------------------------------------------------"
+
```
Comment (LiuYuWei) [inline install/primehub-support:399]: I think we need to output the information about "OOO command not found"
EX: "Kubectl-primehub command not found."

File: install/primehub-support [hunk 3]
```diff
@@ -0,0 +1,407 @@
+#!/bin/bash
+# InfuseAI support data collection script
+HOME=$HOME
+LOG_DIR=$HOME/infuseai_support_logs_$(date +'%Y%m%d%H%M')
```
Comment (LiuYuWei) [inline install/primehub-support:4]: I think we can store the log in some folder. Not in home dir.
EX: 
```LOG_DIR=$HOME/support/infuseai_support_logs_$(date +'%Y%m%d%H%M')```
or
```LOG_DIR=$HOME/log/infuseai_support_logs_$(date +'%Y%m%d%H%M')```

**Review by LiuYuWei — CHANGES_REQUESTED** [review]
Some comments in the script. Please check them. Thanks!

File: install/primehub-support [hunk 4]
```diff
@@ -0,0 +1,407 @@
...
+  echo -e "\033[1;96m$1\033[0m"
+}
+
+mark() {
+  echo -e "\033[0;97m$1\033[0m"
+}
+
+notpresent() {
+  echo -e "\033[0;31m$1\033[0m"
+}
```
[the remark this answers is not in the document]
Comment (huaichehuang) [reply install/primehub-support:54]: I will remove this block for this version, thanks for reminding me.

File: install/primehub-support [hunk 5]
```diff
@@ -0,0 +1,407 @@
...
+	echo "This host is a native Linux Server."
+	echo "Host type match, moving to next step."
+else
+	echo "This host is not a native Linux Server."
+	echo "This script can only be executed on a native GNU/Linux system."
+	exit 1
+fi
+echo ""
+
+# Check if this scrip run by root-like account
```
[the remark this answers is not in the document]
Comment (huaichehuang) [reply install/primehub-support:72]: Noted

File: install/primehub-support [hunk 6]
```diff
@@ -0,0 +1,407 @@
...
+echo "-------------------------------------------------------------------------"
+echo "Check system date and time."
+if [[ "$(command -v ansible)" != "" ]]; then
+	read -p "Please input your inventory file full path, skip this step by pressing enter. (e.g. $HOME/.ansible/inventory.yaml) > " inventory_file
+	if [[ $inventory_file == "" ]]; then
+		echo "Will not run ansible command due to empty input."
+	else
+		ansible -i $inventory_file all -m shell -a 'date' -f99 &> $LOG_DIR/ansible_date_$TIMESTAMP.log
+		echo "System date log is now logged at $LOG_DIR/ansible_date_$TIMESTAMP.log"
+	fi
```
[the remark this answers is not in the document]
Comment (huaichehuang) [reply install/primehub-support:131]: User has to type the full path of their inventory file for ansible command to run, we can check whether the file is existed or not, but if they provide the wrong file, ansible command will still fail.

File: install/primehub-support [hunk 7]
```diff
@@ -0,0 +1,407 @@
...
+SYS_CHECK() {
+echo "Log NVIDIA driver status to $LOG_DIR"
+if [[ "$(command -v nvidia-smi)" == "" ]]; then
+	echo "nvidia-smi command not found, skipping....."
+else
+	echo "nvidia-smi installed, moving to next step."
+	nvidia-smi &> $LOG_DIR/nvidia_driver_$TIMESTAMP.log
+	echo "System NVIDIA driver status is now logged at $LOG_DIR/nvidia_driver_$TIMESTAMP.log"
+fi
+echo ""
```
[the remark this answers is not in the document]
Comment (huaichehuang) [reply install/primehub-support:187]: In this block, the script will check if user has entered inventory file in advance for script to use.
If not, the script will only be running commands on the local system to collect localhost data.

File: install/primehub-support (hunk 1 above)
Comment (huaichehuang) [reply install/primehub-support:243]: As mentioned above, this block is to check whether the local system has ansible command and provided a inventory file.
But I think improvement can be made by removing duplicate conditional check, thanks for the reminder.

File: install/primehub-support (hunk 2 above)
Comment (huaichehuang) [reply install/primehub-support:399]: "OOO command not found" is already shown when you run each block of this script.
If we need a "command not found summary" here, sure I can work on that.

File: install/primehub-support (hunk 3 above)
Comment (huaichehuang) [reply install/primehub-support:4]: That's good advice, cleaner home folder is what we are looking for!
I will create a parent folder for logs to be stored.

File: install/primehub-support [hunk 8]
```diff
@@ -0,0 +1,407 @@
...
+echo "-------------------------------------------------------------------------"
+
+echo ""
+printf "\E[0;90;102m#################################################################\E[0m\n"
+printf "\E[0;90;102m#                                                               #\E[0m\n"
+printf "\E[0;90;102m#  InfuseAI support data collection is completed successfully!  #\E[0m\n"
+printf "\E[0;90;102m#                                                               #\E[0m\n"
+printf "\E[0;90;102m#################################################################\E[0m\n"
+echo ""
+exit 0
```
[the remark this answers is not in the document]
Comment (huaichehuang) [reply install/primehub-support:407]: Sure, it is my next step toward this script. Thanks for reminding me.

File: install/primehub-support (hunk 2 above)
Comment (huaichehuang) [reply install/primehub-support:435]: Create one more log to record "not found" command.

**Review by LiuYuWei — COMMENTED** [review]
1. ceph_rbd_pool 目前只是輸出出來，覺得可以再下一版整理時，想辦法去對照PV，讓這個輸出更容易理解和直接使用(不用此版更改）
2. k8s_status部分，下一版可以改成輸出如下：（可以不用此版更改）
```
The status of all nodes is ready.
The status of all pods is ready.
```
或者以下方式：
```
- The status of nodes 
All nodes are ready.
- The status of pods 
All pods are ready.
```
3. kubernetes_unavailable_deploy
似乎沒有辦法分行
4. ansible 部分在部分環境會出現以下錯誤，可能要再看一下問題怎麼解決會比較好
```
Check system date and time.
timeout: failed to run command ‘read’: No such file or directory
Will not run ansible command due to empty input or non-existent file.
```

**Pull request merged**